file/index.js

```diff
@@ -0,0 +1,120 @@
+                continue;
+            }
+
+            modifiedFiles.push(result.Contents[index]);
+            
+            // If we want to process limited number of files in one lambda execution, so we should only add those number of files
+            if (modifiedFiles.length === limitNumberOfFileToUpdateInOneRun) {
+                response.hasMore = true;
+                break;
+            }
+        }
+    }).catch(function (err) {
+        callback(err); // Return when some error occurred while listing objects in bucket
+        console.log("Error in listing bucket", JSON.stringify(err) + "\n");
+    });
+
+    // Waiting for listing of objects so we can get all modified files
+    await listObjectsPromise;
+
+    // Sort in ascending order
+    modifiedFiles.sort(function (a, b) { return (a.LastModified > b.LastModified) ? 1 : ((b.LastModified > a.LastModified) ? -1 : 0); });
+
+    // Process files one by one
+    for (var index = 0; index < modifiedFiles.length; index++) {
+        let modifiedFile = modifiedFiles[index];
+
+        let params = {
+            Bucket: bucketName,
+            Key: modifiedFile.Key
+        };
+
+        let getObjectPromise = s3.getObject(params).promise();
+
+        getObjectPromise.then(function (result) {
+            let fileData = result.Body.toString('utf-8');
+            var rows=fileData.split("\n");
+            var headers;
+            var count=0;
+            
+            for(var i in rows){
+                if(rows[i].trim().length===0)continue;
+                count++;
+                var cols=rows[i].split("\t");
+                // Extract headers
+                if(count===1){
+                    headers=cols;
+                    continue;
+                }
+                
+                var obj={};
+                for(var j in cols){
+                    obj[headers[j]]=cols[j];
+                }
+                if (modifiedFile.Key.endsWith("_delete.tsv")) {
+                    response.delete.near_earth_objects.push(obj);
+                } else{
+                    response.insert.near_earth_objects.push(obj);
+                }
+            }
+        }).catch(function (err) {
+            callback(err); // Return when some error occurred while reading any file
+            console.log("Error in getting object : " + err + "\n");
+        });
+
+        await getObjectPromise;
+
+        // Same last modified of processed file so we process files after that in next run
+        response.state.since = modifiedFile.LastModified;
+    }
+
+    // Once response in generated use callback to finish lambda execution with response
+    callback(null, response);
+}
+
+function initializeResponse(state) {
+    // Don't assign the value directly, it means you are assigning as a reference, 
+    // Now state of response doesn't get affected when change state variable in method
+    return {
+        "state": {...state}, 
```

Comment:
it's weird way to declare js objects, usually you don't put field names in quotes. While you will have to do that for `delete` I believe, it's a reserved word (it's sad that we used this name in the first place).